src/main/java/org/openrewrite/java/migrate/apache/commons/lang/ApacheCommonsFileUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.apache.commons.lang;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Arrays;
+
+public class ApacheCommonsFileUtils {
+    private static class Write {
+        @BeforeTemplate
+        void before(File file, CharSequence data, Charset cs) throws Exception {
+            FileUtils.write(file, data, cs);
+        }
+
+        @AfterTemplate
+        void after(File file, CharSequence data, Charset cs) throws Exception {
+            Files.write(file.toPath(), Arrays.asList(data), cs);
```

Comment:
@knutwannheden (no rush! just interesting)
This fails compile in the generated recipe due to
```
java: reference to compile is ambiguous
both method compile(org.openrewrite.java.JavaVisitor<?>,java.lang.String,org.openrewrite.java.JavaTemplate.P3<?,?,?>) in org.openrewrite.java.JavaTemplate
 and method compile(org.openrewrite.java.JavaVisitor<?>,java.lang.String,org.openrewrite.java.JavaTemplate.F3<?,?,?,?>) in org.openrewrite.java.JavaTemplate match
```

Reply:
Any idea what might be going on here? The generated code is:
```java
final JavaTemplate before = JavaTemplate.compile(this, "before", (java.io.File file, CharSequence data, java.nio.charset.Charset cs) -> FileUtils.write(file, data, cs)).build();
final JavaTemplate after = JavaTemplate.compile(this, "after", (java.io.File file, CharSequence data, java.nio.charset.Charset cs) -> Files.write(file.toPath(), Arrays.asList(data), cs)).build();
```

Reply:
We can work around this with a simple cast of course. ;/

Reply:
Haven't looked at it yet. Possibly we need to generate an explicit cast, because the return type of the called method is not void like that of the template method.